packages/react/src/components/CopyButton/CopyButton.tsx

```diff
@@ -6,13 +6,57 @@
  */
 
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { MouseEventHandler } from 'react';
 import classnames from 'classnames';
 import { Copy as CopyIcon } from '@carbon/icons-react';
+import { ButtonProps } from '../Button';
 import Copy from '../Copy';
 import { LayoutConstraint } from '../Layout';
 import { usePrefix } from '../../internal/usePrefix';
 import { noopFn } from '../../internal/noopFn';
+
+export interface CopyButtonProps extends ButtonProps<'button'> {
```

Comment:
I have changed this to extend Carbon's ButtonProps rather than the native button.  I have done this because CodeSnippet passes a size prop to CopyButton, which gets used by Button down the line.  Should i also change the Copy type to extend ButtonProps?